MENTAL-MODEL.md

```diff
@@ -0,0 +1,376 @@
+
+Avoiding the secure endpoint requirement is convenient for testing.
+
+```rs
+let client = reqwest::blocking::Client::builder()
+    .danger_accept_invalid_certs(danger_accept_invalid_certs)
+    .build()
+    .with_context(|| "Failed to build reqwest http client")?;
+let response = client
+    .post(req.url)
+    .body(req.body)
+    .header("Content-Type", "text/plain")
+    .send()
+    .with_context(|| "HTTP request failed")?;
+```
+
+### 6. Process the response
+
+An `Ok` response should include a Payjoin Proposal PSBT. Check that it's signed, following protocol, not trying to steal or otherwise error.
+
+```rs
+// TODO display well-known errors and log::debug the rest
+let psbt = ctx.process_response(response).with_context(|| "Failed to process response")?;
+log::debug!("Proposed psbt: {:#?}", psbt);
+```
+
+Payjoin response errors (called [receiver's errors in spec](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#receivers-well-known-errors)) come from a remote server and can be used to "maliciously to phish a non technical user." Only those listed as "well known" in the spec should be displayed with preset messages to prevent phishing.
+
+### 7. Sign and finalize the Payjoin Proposal PSBT
+
+Most software can handle adding the last signatures to a PSBT without issue.
+
+```rs
+let psbt = bitcoind
+    .wallet_process_psbt(&serialize_psbt(&psbt), None, None, None)
+    .with_context(|| "Failed to process PSBT")?
+    .psbt;
+let tx = bitcoind
+    .finalize_psbt(&psbt, Some(true))
+    .with_context(|| "Failed to finalize PSBT")?
+    .hex
+    .ok_or_else(|| anyhow!("Incomplete PSBT"))?;
+```
+
+### 8. Broadcast the Payjoin Transaction
+
+In order to preserve privacy between the transaction and the IP address from which it originates, transaction broadcasting should be done using Tor, a VPN, or proxy.
+
+```rs
+let txid =
+    bitcoind.send_raw_transaction(&tx).with_context(|| "Failed to send raw transaction")?;
+log::info!("Transaction sent: {}", txid);
+```
+
+📤 Sending payjoin is just that simple.
+
+## Receive a Payjoin
+
+The `receiver` feature provides all of the check methods, PSBT data manipulation, coin selection, and transport structures to receive payjoin and handle errors in a privacy preserving way.
+
+Receiving payjoin entails listening to a secure http endpoint for inbound requests. The endpoint is displayed in the `pj` parameter of a bip21 request URI.
+
+The reference implementation uses `rouille` sync http server and Bitcoin Core RPC.
+
+```rs
+fn receive_payjoin(
+    bitcoind: bitcoincore_rpc::Client,
+    amount_arg: &str,
+    endpoint_arg: &str,
```

Comment:
endpoint displayed in the BIP 21 URI could be configured rather than passed as an argument, and the amount should be optional https://github.com/payjoin/rust-payjoin/issues/54